feat(session modal): introduce session bookmarking

spoons-and-mirrors opened the pull request:
Building on https://github.com/sst/opencode/pull/1821


#### With pins
<img width="646" height="283" alt="image" src="https://github.com/user-attachments/assets/1f5ac1c4-8f41-4b66-9e59-0a947f5d96e1" />

#### Pinned sessions only
<img width="637" height="158" alt="image" src="https://github.com/user-attachments/assets/020c40c2-b70c-4c35-85b9-10e3eff69546" />

spoons-and-mirrors committed:
feat: add session rename functionality to TUI modal

Enable users to rename sessions directly from the session modal by pressing 'r' key.

- Add PATCH /session/:id API endpoint for updating session title
- Extend Go SDK with UpdateSession method and SessionUpdateParams type
- Add UpdateSession method to TUI app for API integration
- Implement rename mode in session modal with textinput component
- Update modal help text to show 'r rename' option
- Handle Enter/Esc for confirm/cancel rename operations

🤖 Generated with [opencode](https://opencode.ai)

Co-Authored-By: opencode <noreply@opencode.ai>

spoons-and-mirrors committed:
feat: update modal title to "Rename Session" during rename mode

Change the session modal title from "Switch Session" to "Rename Session" when user enters rename mode, providing clearer context for the current operation.

🤖 Generated with [opencode](https://opencode.ai)

Co-Authored-By: opencode <noreply@opencode.ai>

spoons-and-mirrors committed:
fix: ESC key in rename mode returns to session list instead of closing modal

When in session rename mode, pressing ESC now exits rename mode and returns to the session list view instead of closing the modal entirely. This provides a more intuitive user experience.

- Add ReopenSessionModalMsg to handle rename mode ESC behavior
- Modify sessionDialog.Close() to detect rename mode and reopen modal
- Add TUI handler for ReopenSessionModalMsg to recreate session modal
- Remove explicit ESC handling from rename mode (now handled by Close method)

🤖 Generated with [opencode](https://opencode.ai)

Co-Authored-By: opencode <noreply@opencode.ai>

spoons-and-mirrors committed:
feat(session): introduce pins
- `p` to pin session in the session modal
- `shift+p` to see pinned session only

spoons-and-mirrors committed:
fix: pinning does reorder the list. selecting a session puts it at the top. changing the ugly pin for a dot with success color.

spoons-and-mirrors committed:
guidelines: prefer single word variable where possible

thdxr commented:
lol ignore these reviews they're garbage

adamdotdevin commented:
love the idea of pinning sessions, but a few UX/visual nits:
- i don't love the emojis, can we indicate pins another way?
- that bottom menu is getting a bit crowded; can we just always show with pins at the top, do we need to toggle?

spoons-and-mirrors commented:
yeah sorry I didn't update after my latest commit, I  didn't like the pins either and moved to a dot with success color (but had some troubles with the background when highlighting, showing a black box...)

anywho, it now looks like that
<img width="655" height="205" alt="image" src="https://github.com/user-attachments/assets/e48668b1-6958-431f-8241-3b111ff13580" />

I realize that pinning is actually a miss-use of the term on my part, I was more thinking of a "bookmark" system.
I feel it doesn't make sense to push the pins/bookmark at the top of the session list, but rather leave them where they are in the item list. This is why I think the bookmark view is a good addition, to filter and only show our bookmarked sessions. It feels more organized this way rather than constantly pushing things at the top **when** we mark them.

Example:
You make a few research sessions and move on to something else. Then later, you go and bookmark those research sessions because they had insight you will need at some point, **but not now**. It doesn't make sense to me to push them at the top, they will just be a nuisance. Plus if you have say 10 pins, you will have to scroll to see your working sessions.

Regarding the footer being too crowded, we could maybe change:
- `n new session` -> `n new`
- `P pinned view` > `P filter`
and maybe add 2 spaces between shortcuts:
`n new  r rename  p pin  P filter`
or 3 
`n new   r rename   p pin   P filter` (I like 3 better)
we could also use a warning color for the letter shortcut adding some dynamism to the footer?

The highlight issue I mentioned is this
<img width="450" height="199" alt="image" src="https://github.com/user-attachments/assets/716464ae-fdac-4ff3-84b7-b85c6c974163" />
EDIT: fixed!
<img width="648" height="203" alt="image" src="https://github.com/user-attachments/assets/9c9b54a0-9e58-480b-acba-31f3ce43b6e1" />


PS: maybe `TAB` is a better way of toggling to the filtered view?

spoons-and-mirrors committed:
tui: show only active dot for current session, update shortcut help (warning color, spacing, labels) and bind bookmarks to TAB

spoons-and-mirrors commented:
Hey @adamdotdevin, here's how the suggestions look like

<img width="646" height="280" alt="image" src="https://github.com/user-attachments/assets/69ac41be-0380-419b-bc97-25e5692d0de6" />

or

<img width="646" height="33" alt="image" src="https://github.com/user-attachments/assets/f3b45b9a-44dd-4f12-91db-72bdcc2bbe06" />

spoons-and-mirrors committed:
fixes: renaming pin to bookmark for consistency, looking up sessions by id instead of title

spoons-and-mirrors committed:
fixing background color on hover

spoons-and-mirrors committed:
renaming header in bookmark view

spoons-and-mirrors committed:
fixing text appearing in dim/muted when session is bookmarked
